internal/controller/phases.go

```diff
@@ -554,42 +552,6 @@ func (p *phaseReconciler) newClusterClient() cluster.Client {
 	}))
 }
 
-// repositoryFactory returns the repository implementation corresponding to the provider URL.
-// inspired by https://github.com/kubernetes-sigs/cluster-api/blob/124d9be7035e492f027cdc7a701b6b179451190a/cmd/clusterctl/client/repository/client.go#L170
-func repositoryFactory(ctx context.Context, providerConfig configclient.Provider, configVariablesClient configclient.VariablesClient) (repository.Repository, error) {
-	// parse the repository url
-	rURL, err := url.Parse(providerConfig.URL())
-	if err != nil {
-		return nil, fmt.Errorf("failed to parse repository url %q", providerConfig.URL())
-	}
-
-	if rURL.Scheme != httpsScheme {
-		return nil, fmt.Errorf("invalid provider url. there are no provider implementation for %q schema", rURL.Scheme)
-	}
-
-	// if the url is a GitHub repository
-	if rURL.Host == githubDomain {
```

Comment:
I think these constants can be removed now
```

	httpsScheme             = "https"
	githubDomain            = "github.com"
	gitlabHostPrefix        = "gitlab."
	gitlabPackagesAPIPrefix = "/api/v4/projects/"
```

Reply:
done!